src/MerchantCenter/MerchantStatuses.php

```diff
@@ -652,4 +651,29 @@ protected function get_issue_severity( array $row ): string {
 
 		return $is_warning ? self::SEVERITY_WARNING : self::SEVERITY_ERROR;
 	}
+
+	/**
+	 * Returns an array of Google Product IDs associated with all synced WooCommerce products.
+	 * Note: excludes variable parent products as only the child variation products are actually synced
+	 * to Merchant Center
+	 *
+	 * @return array Google Product IDS
+	 */
+	protected function get_synced_google_ids(): array {
+		/** @var ProductRepository $product_repository */
+		$product_repository = $this->container->get( ProductRepository::class );
+
+		// Don't include variable parent products as they aren't actually synced to Merchant Center.
+		$args['type']        = array_diff( ProductSyncer::get_supported_product_types(), [ 'variable' ] );
+		$synced_product_ids  = $product_repository->find_synced_product_ids( $args );
+		$google_ids_meta_key = $this->prefix_meta_key( ProductMetaHandler::KEY_GOOGLE_IDS );
+		$synced_google_ids   = [];
+		foreach ( $synced_product_ids as $product_id ) {
+			$meta_records = get_post_meta( $product_id, $google_ids_meta_key );
+			foreach ( $meta_records as $google_ids ) {
+				$synced_google_ids = array_merge( $synced_google_ids, array_values( $google_ids ) );
+			}
```

Comment:
We only store the google ID's once per post, so we can set `$single = true` in the call to `get_post_meta`, which eliminates an extra foreach loop.

Are we ever worried that the `$google_ids` value is not going to return an array? It should only happen if DB data is garbled, but that's not unheard of. Maybe we should add an extra safety check here.

Reply:
Nice, made the changes in 136c764966baaf38d879155d8497fefadbcf604d and 3f773778f47b1630e7cff90dc66fa278883260dc